packages/backend-defaults/src/entrypoints/rootHttpRouter/http/MiddlewareFactory.ts

```diff
@@ -137,18 +160,27 @@ export class MiddlewareFactory {
    * @returns An Express request handler
    */
   logging(): RequestHandler {
-    const logger = this.#logger.child({
-      type: 'incomingRequest',
-    });
-    const customMorganFormat =
-      '[:date[clf]] ":method :url HTTP/:http-version" :status :res[content-length] ":referrer" ":user-agent"';
-    return morgan(customMorganFormat, {
-      stream: {
-        write(message: string) {
-          logger.info(message.trimEnd());
+    const logger = this.#logger;
+    return morgan(
+      (tokens: TokenIndexer, req: IncomingMessage, res: ServerResponse) => {
+        const meta = getLogMeta(tokens, req, res);
+        const message = getLogMessage(tokens, req, res);
+        return JSON.stringify({ meta, message });
+      },
+      {
+        stream: {
+          write(json: string) {
+            const { meta, message } = JSON.parse(json);
+            logger.info(message.trimEnd(), {
+              type: 'incomingRequest',
+              ...Object.entries(meta).reduce((rest, [key, value]) => {
```

Comment:
it's actually the case that `JSON.stringify` throws away keys that have an undefined value, so you can skip the entire reduce.

Reply:
Ah-a, fixed 🙏🏻